File: data/src/main/java/com/hgianastasio/biblioulbrav2/data/disk/GenericCache.java
package com.hgianastasio.biblioulbrav2.data.disk;

import java.io.IOException;

/**
 * Created by heitorgianastasio on 4/28/17.
 */

public interface GenericCache<T> {
    T get() throws IOException;
    boolean save(T t) throws IOException;
    boolean clearCache();
    boolean isUpdated();
}

File: data/src/main/java/com/hgianastasio/biblioulbrav2/data/disk/GenericCache.kt
package com.hgianastasio.biblioulbrav2.data.disk

import java.io.IOException

/**
 * Created by heitorgianastasio on 4/28/17.
 */
interface GenericCache<T> {
    @Throws(IOException::class)
    fun get(): T

    @Throws(IOException::class)
    fun save(t: T): Boolean
    fun clearCache(): Boolean
    val isUpdated: Boolean
}